### PR TITLE
Deleting a mapping within a struct vulnerability added.

### PR DESCRIPTION
**Pull request by 0xSandyy**
## Related Issue

- Closes: https://github.com/kadenzipfel/smart-contract-vulnerabilities/issues/73

## Checklist

- [x] I have read and followed the [style guide](https://github.com/kadenzipfel/smart-contract-vulnerabilities/blob/master/style-guide.md)

### Describe the changes you've made:
This contribution adds a vulnerability that deals with an unique behavior caused by ``deleting`` structs which have a mapping inside them. ``delete`` gives a common idea that it will delete every content inside but this doesn't always happen and this contribution expands on that.

## Type of change

Select the appropriate checkbox:

- [ ] Bug fix (fixing an issue with existing vulnerability data)
- [x] New feature (adding a new vulnerability or category)
- [ ] Documentation update (improving existing information)

## Additional Information
The impact of this issue is not straightforward but having data that is supposed to be deleted can have unexpected consequences.

### CODE REVIEW

**Comment by 0xSandyy** [comment]
Hi, the issue is in the latest commit i.e ``4992e34``. I will make sure to add more clear commit messages.

File: vulnerabilities/mapping-within-struct.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Deleting a mapping within a struct
```
Comment (kadenzipfel) [inline vulnerabilities/mapping-within-struct.md:1]: Can we capitalize the title to "Deleting a Mapping Within a Struct"

**Review by kadenzipfel — CHANGES_REQUESTED** [review]
It might also be useful to explain the reasoning why the dynamic data doesn't get removed. IIUC it's just a pointer to that dynamic storage which gets deleted

**Comment by 0xSandyy** [comment]
Hi, as per  the docs and some research:
> Mappings are implemented as hash tables and the Ethereum Virtual Machine (EVM) does not keep track of which keys have been used in the mapping. As a result, EVM doesn't know how to "reset" a mapping. 
However, individual keys and what they map to can be deleted: If a is a mapping, then delete a[x] will delete the value stored at x.

File: vulnerabilities/mapping-within-struct.md (hunk 1 above)
Comment (0xSandyy) [reply vulnerabilities/mapping-within-struct.md]: Fixed in the latest commit!

**Review by kadenzipfel — APPROVED** [review]

**Pull request merged**